File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Deque<Character> left = new LinkedList<>();
        
        for(char c : s.toCharArray()) {
            if(left.isEmpty()) {
                if(c == ')'){
                    return false;
                }
                
                left.offerLast(c);
            }else {
                if(c == ')') {
                    left.pollLast();
                }else {
                    left.offerLast(c);
                }
            }
        }
        
        if(left.size() == 0){
            return true;
        }
        
        return false;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    const list = [];
    
    for(let i of s){
        if(i === ")") {
            if(list.length === 0) {
                return false;
            }
            
            list.pop();
        }else {
            list.push(i);
        }
    }
    
    if(list.length === 0) {
        return true
    }

    return false;
}